components/raftstore/src/store/fsm/peer.rs

```diff
@@ -631,31 +631,32 @@ where
                         .propose
                         .request_wait_time
                         .observe(duration_to_sec(cmd.send_time.saturating_elapsed()) as f64);
+
                     if let Some(Err(e)) = cmd.extra_opts.deadline.map(|deadline| deadline.check()) {
                         cmd.callback.invoke_with_response(new_error(e.into()));
                         continue;
                     }
 
-                    let req_size = cmd.request.compute_size();
                     if self.ctx.cfg.cmd_batch
-                        && self.fsm.batch_req_builder.can_batch(&self.ctx.cfg, &cmd.request, req_size)
                         // Avoid to merge requests with different `DiskFullOpt`s into one,
                         // so that normal writes can be rejected when proposing if the
                         // store's disk is full.
                         && ((self.ctx.self_disk_usage == DiskUsage::Normal
                             && !self.fsm.peer.disk_full_peers.majority())
                             || cmd.extra_opts.disk_full_opt == DiskFullOpt::NotAllowedOnFull)
+                        && let req_size = cmd.request.compute_size()
```

Comment:
Weird style.

Reply:
I searched around and found several similar cases in rust-clippy:
https://github.com/rust-lang/rust-clippy/blob/f7e2cb4470f5288e6657166fb0fcdf4b7043517f/clippy_lints/src/dereference.rs#L755
https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/manual_non_exhaustive.rs#L130

Reply:
It doesn't justify it's a weird style. Putting pattern matching that actually just does assignment to bool field doesn't make sense to me.

Reply:
As I explained in https://github.com/tikv/tikv/pull/13230#discussion_r940047751, I presume you treat this particular `let` differently because it can never fail. But pattern matching is never a boolean expression. Thinking it is is a misperception.

Reply:
> you treat this particular let differently because it can never fail.

I treat it differently because it "just does assignment".

Reply:
Fine, I'll keep it simple to reduce friction. But for the record I didn't expect so much push-back from this line..